### DISCUSSION TITLE
Introduce "pass" keyword

### DISCUSSION THREAD

**Discussion by Audionysos** [dotnet/csharplang#7709] [Language Ideas]
A "pass" keyword would specify a property which only purpose is to be a shortcut to a member of some other object.

For example, instead of writing this:

``` 
public class Button {
	public ButtonStates states { get; set; } = new ButtonStates();
	
	public object? content {
		get => states[0].content;
		set => states[0].content = value;
	}
	public SurfacePlacement placement {
		get => states[0].placement;
		set => states[0].placement = value;
	}
	public DimensionsData size {
		get => states[0].size;
		set => states[0].size = value;
	}
}
```

You would write this:

```
public class Button {
	public ButtonStates states { get; set; } = new ButtonStates();

	pass content => states[0].content;
	pass placement => states[0].placement;
	pass size => states[0].size;
}
```
Or if you want the same name just: 

```
pass => states[0].content;
pass => states[0].placement;
pass => states[0].size;
```

I would suggest that the accessors would be public as default (private pass would make no sense, maybe besides making abbreviations...) and you could change it like this:

```
pass  { get; internal set; } => states[0].content; //public `get` and internal `set` accessor
pass { get; } => states[0].placement;  //only public 'get accessor
pass => states[0].size; // bot public get and set
```

Of course the accessors would be made available only if the target type exposes them  - if the button state class looks like this:
```
public class ButtonState {
	public object? content { get; set; }
	public SurfacePlacement placement { get; set; } = new();
	public DimensionsData size { get; } = new();
}
```

the `pass => states[0].size;` would only create the `get` accessor.
I think you get the idea.

**Comment by En3Tho** [comment]
This can be achieved with a source generator. I don't think you need a keyword for this.

**Comment by HaloFour** [comment]
See: #1007
